genai-perf/genai_perf/profile_data_parser/llm_profile_data_parser.py

```diff
@@ -204,6 +204,8 @@ def _get_input_token_count(self, req_inputs: dict) -> int:
         """Deserialize the request input and return tokenized inputs."""
         if self._service_kind == "triton":
             input_text = req_inputs["text_input"]
+        elif self._service_kind == "triton_c_api":
+            return len(req_inputs["input_ids"])  # no tokenizer required
```

Comment:
This is specific to TRTLLM since TRTLLM is the only model that we support through triton c api.

Reply:
I see. I won't block on this if that's the approach that you, @matthewkotila, and @debermudez are using. It just seems misleading to make it look like this is the logic for supporting C API, but it's really just TRT-LLM via C API and our logic is exclusively designed for TRT-LLM (on the metrics side, not just the parsing side).

I would personally rather [this line](https://github.com/triton-inference-server/perf_analyzer/pull/33/files#diff-a056e8717cfab56e4f62b45b9b35efced45190c204f05ef8eea56a19bd7623e9R50) not exist and we just pass C API directly to the PA command. Then, it's clear in this file that we're only using this logic for the tensorrtllm_engine service-kind.

Reply:
Update: I think I didn't need to tag Matt or Elias here, apologies for that. I was thinking you all worked on the designs for the TRT-LLM engine path, but I'm realizing that this is simply a code question. It's not that we're adding a C API service-kind, just that we're overriding the TRT-LLM engine service-kind with C API behind the scenes (in the linked code), which led to this code being confusing on my end. I still think it could be good to not do that overriding to make the code clearer.

Reply:
sorry could you rephrase your question? I don't think I quite understood what your suggestion was here.

Reply:
I think David is retracting his suggestion?

Reply:
+ I think what will solve a lot of confusion is another refactor in profile data parser classes so that we have one classes for each endpoints (e.g. openai chat completion, triton vllm, etc). This way, we don't have these nasty if-else blocks.

Reply:
Discussed offline. PA requires service-kind to be triton_c_api and GenAI-Perf uses tensorrtllm_engine. To help with clarity, Hyunjae offered to push changes to use tensorrtllm_engine as the service-kind across GenAI-Perf and just pass in triton_c_api for PA where it needs it.

Reply:
Looks like the tensorrtllm_engine part is from PA's profile export and cannot be changed. Resolving.